modules/tinymce/src/plugins/emoticons/test/ts/browser/DifferentEmojiDatabaseTest.ts

```diff
@@ -14,44 +17,36 @@ const getFilename = (url: string) => {
   return '';
 };
 
-UnitTest.asynctest('browser.tinymce.plugins.emoticons.DifferentEmojiDatabaseTest', (success, failure) => {
-  EmoticonsPlugin();
-  SilverTheme();
-
-  const sTestEditorWithSettings = (categories, databaseUrl) => Step.async((onStepSuccess, onStepFailure) => {
-    TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-      const tinyApis = TinyApis(editor);
-      const tinyUi = TinyUi(editor);
+describe('browser.tinymce.plugins.emoticons.DifferentEmojiDatabaseTest', () => {
+  before(() => {
+    Plugin();
+    Theme();
+  });
 
-      Pipeline.async({}, [
-        tinyApis.sFocus(),
-        tinyUi.sClickOnToolbar('click emoticons', 'button'),
-        Chain.asStep({}, [
-          tinyUi.cWaitForPopup('wait for popup', 'div[role="dialog"]')
-        ]),
-        Waiter.sTryUntil(
-          'Wait for emojis to load',
-          UiFinder.sNotExists(SugarBody.body(), '.tox-spinner')
-        ),
-        Chain.asStep(SugarBody.body(), [
-          UiFinder.cFindAllIn('[role="tab"]'),
-          Chain.mapper((elements: SugarElement[]) => Arr.map(elements, (elm: SugarElement) => elm.dom.textContent)),
-          Assertions.cAssertEq('Categories match', categories)
-        ])
-      ], onSuccess, onFailure);
-    }, {
+  const pTestEditorWithSettings = async (categories: string[], databaseUrl: string) => {
+    const editor = await McEditor.pFromSettings<Editor>({
       plugins: 'emoticons',
       toolbar: 'emoticons',
-      theme: 'silver',
       base_url: '/project/tinymce/js/tinymce',
       emoticons_database_url: databaseUrl,
       emoticons_database_id: 'tinymce.plugins.emoticons.' + getFilename(databaseUrl)
-    }, onStepSuccess, onStepFailure);
-  });
+    });
 
-  Pipeline.async({},
-    Log.steps('TBA', 'Emoticon: Loading databases from different urls', [
-      sTestEditorWithSettings([ 'All', 'People' ], '/project/tinymce/src/plugins/emoticons/test/js/test-emojis.js'),
-      sTestEditorWithSettings([ 'All', 'Travel and Places' ], '/project/tinymce/src/plugins/emoticons/test/js/test-emojis-alt.js')
-    ]), success, failure);
+    TinyUiActions.clickOnToolbar(editor, 'button');
+    await TinyUiActions.pWaitForPopup(editor, 'div[role="dialog"]');
+    await Waiter.pTryUntil(
+      'Wait for emojis to load',
+      () => UiFinder.notExists(SugarBody.body(), '.tox-spinner')
```

Comment:
[not a thing to change] in the past I've always tried messing with aria roles to figure out if everything had loaded. I like this idea a lot